### PR TITLE
Add Restart command and keyboard shortcut for moving time to start of timeline

### PR DESCRIPTION
**Pull request by h3mosphere**
### WHAT
Add Restart button to timeline UI. This sets the timeline back to zero.
​
### WHY
Replaying from the start is helpful. Looping is the only other way to do this without clicking on the timeline directly.

### HOW
?

Note: I don't really like that it is called Restart. Reset probably is better, but that is taken by another function (ResetViewer), and causes some name clashes with icon constants etc. Any naming ideas welcome.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [ ] I've included a screenshot or gif (if applicable)

### CODE REVIEW

File: crates/re_viewer/src/app.rs [hunk 1]
```diff
@@ -351,6 +353,7 @@ impl App {
             TimeControlCommand::TogglePlayPause => {
                 time_ctrl.toggle_play_pause(times_per_timeline);
             }
+            TimeControlCommand::Restart => time_ctrl.set_time(0),
```
Comment (h3mosphere) [inline crates/re_viewer/src/app.rs:356]: Not sure if setting to 0 will work with all time types?

**Comment by h3mosphere** [comment]
Hey.

I'm just looking at the code base mostly. It seemed like a nice feature, but I understand your position.

Why? I am mostly running short simulations, and its makes sense in this scenario to be able to quickly go back to the start.

It only took half an hour or so, so no worries if it is out of scope. I am having fun playing with rerun. And I'll discuss more on discord from now one.

Thanks

**Comment by nikolausWest** [comment]
Perhaps it makes sense to solve this by instead revisiting the keyboard shortcuts for play control? I don't remember what our current controls allow with keyboard shortcuts even so that should probably be added to rerun.io/docs as well

**Comment by nikolausWest** [comment]
> Perhaps it makes sense to solve this by instead revisiting the keyboard shortcuts for play control? I don't remember what our current controls allow with keyboard shortcuts even so that should probably be added to rerun.io/docs as well

@emilk: thoughts?

File: crates/re_viewer/src/app.rs (hunk 1 above)
Comment (emilk) [reply crates/re_viewer/src/app.rs:356]: It won't - it should go back to the first time point of the current timeline. You can use `times_per_timeline` with `time_ctrl.timeline()` to figure that out!

**Comment by emilk** [comment]
The restart button looks like a mirror image of the follow-button, but they act very differently, which is confusing. In particular, the play-, follow-, and pause-buttons are all states (clicking them enters the play/follow/pause-state), but "restart" is not:

restart


I'd vote for adding this as a command (available as a keyboard shortcut and via the command pallet) but hold off on adding it as a GUI button, as that requires more design work imho

File: crates/re_ui/src/command.rs [hunk 2]
```diff
@@ -181,6 +183,7 @@ impl Command {
             Command::ToggleCommandPalette => Some(cmd(Key::P)),
 
             Command::PlaybackTogglePlayPause => Some(key(Key::Space)),
+            Command::PlaybackRestart => Some(ctrl_shift(Key::Space)),
```
Comment (emilk) [inline crates/re_ui/src/command.rs:186]: Is there some prior art for what is a good key combination?

On my Mac I find:
* Spotify: `Cmd Left`
* Logic Pro: `Return`
* VLC: `Cmd .`

**Review by emilk — CHANGES_REQUESTED** [review]

**Comment by h3mosphere** [comment]
> The restart button looks like a mirror image of the follow-button, but they act very differently, which is confusing. In particular, the play-, follow-, and pause-buttons are all states (clicking them enters the play/follow/pause-state), but "restart" is not:
> 
> restart [ restart ](https://user-images.githubusercontent.com/1148717/231135259-2f7b1721-2ea9-49c4-a1d8-633dfb528383.gif) [ ](https://user-images.githubusercontent.com/1148717/231135259-2f7b1721-2ea9-49c4-a1d8-633dfb528383.gif)
> 
> I'd vote for adding this as a command (available as a keyboard shortcut and via the command pallet) but hold off on adding it as a GUI button, as that requires more design work imho

Yes, I just flipped the image, without thinking too much about it, I should have mentioned that this was a placeholder.. 

Adding as a command is perfect for me. I'll remove the UI code, and change the command key code to ctrl(Key::ArrowLeft)

File: crates/re_viewer/src/misc/time_control.rs [hunk 3]
```diff
@@ -293,6 +293,14 @@ impl TimeControl {
         }
     }
 
+    pub fn restart(&mut self, times_per_timeline: &TimesPerTimeline) {
```
Comment (h3mosphere) [inline crates/re_viewer/src/misc/time_control.rs:296]: I copied this from toggle_play_pause(), hopefully the logic is right..

**Comment by nhemsley** [comment]
I made those changes @emilk , but It's not letting me mark as resolved..

**Review by emilk — APPROVED** [review]

**Review by emilk — APPROVED** [review]
Works great - thanks for your contribution!

**Pull request merged**